Jenkinsfile

```diff
@@ -30,84 +31,129 @@ properties([
+                    engine/build/resources/main/org/terasology/engine/version/versionInfo.properties,
+                    natives/**,
+                    build-logic/src/**,
+                    build-logic/*.kts
+                '''
+            }
         }
-    }
 
-    stage('Publish') {
-        if (specialBranch) {
-            withCredentials([usernamePassword(credentialsId: 'artifactory-gooey', usernameVariable: 'artifactoryUser', passwordVariable: 'artifactoryPass')]) {
-                sh './gradlew --console=plain -Dorg.gradle.internal.publish.checksums.insecure=true publish -PmavenUser=${artifactoryUser} -PmavenPass=${artifactoryPass}'
+        stage('Unit Tests') {
+            steps {
+                sh './gradlew --console=plain unitTest'
+            }
+            post {
+                always {
+                    // Gradle generates both a HTML report of the unit tests to `build/reports/tests/*`
+                    // and XML reports to `build/test-results/*`.
+                    // We need to upload the XML reports for visualization in Jenkins.
+                    //
+                    // See https://docs.gradle.org/current/userguide/java_testing.html#test_reporting
+                    junit testResults: '**/build/test-results/unitTest/*.xml'
+                }
             }
-        } else {
-            println "Running on a branch other than 'master' or 'develop' bypassing publishing"
         }
 
-        // Trigger the Omega dist job to repackage a game zip with modules
-        if (env.JOB_NAME.equals("Terasology/engine/develop")) {
-            build job: 'Terasology/Omega/develop', wait: false
-        } else if (env.JOB_NAME.equals("Terasology/engine/master")) {
-            build job: 'Terasology/Omega/master', wait: false
-        } else if (env.JOB_NAME.equals("Nanoware/Terasology/develop")) {
-            build job: 'Nanoware/Omega/develop', wait: false
-        } else if (env.JOB_NAME.equals("Nanoware/Terasology/master")) {
-            build job: 'Nanoware/Omega/master', wait: false
+        stage('Publish') {
+            when {
+                expression {
+                    specialBranch
+                }
+            }
+            steps {
+                withCredentials([usernamePassword(credentialsId: 'artifactory-gooey', \
+                                                    usernameVariable: 'artifactoryUser', \
+                                                    passwordVariable: 'artifactoryPass')]) {
+                    sh '''./gradlew
+                        --console=plain
+                        -Dorg.gradle.internal.publish.checksums.insecure=true
+                        publish
+                        -PmavenUser=${artifactoryUser}
+                        -PmavenPass=${artifactoryPass}
+                    '''
+                }
+                script {
+                    // Trigger the Omega dist job to repackage a game zip with modules
```

Comment:
Was not this inside the `specialBranch` conditional before? I agree it should be!

I'm not sure if there's a disadvantage to using a `script` block like this. I'm okay with assuming it's fine until we learn otherwise.

Reply:
The problem I faced here is that ther is this `when` clause to only run a stage if a specific condition is met, but from what I found this is only available for whole stages, not for individual steps.
The alternative would be to create four different stages, each with the matching `when expression`.
I figured that with the script block we avoid the duplication and additional stages shown in the ui (of which at max one will ever be executed in a pipeline run).

Reply:
And no, this was outside of the `specialBranch` conditional also before. Directly after it, to be precise.